### ISSUE TITLE
Feature Request: Option to show ADs

### ISSUE THREAD

**Issue by alastairlundy** [futo-org/grayjay-android#815]
You may think I'm crazy in wanting this as an option but in some instances I would like to support creators by having the YT ADs shown that would normally appear when watching a video or stream on the YT app.

If this isn't possible with watching YouTube with the current video player for some reason, here's a potential hypothetical solution: when this option is enabled the player could be replaced with an embedded web view with the YT video open and playing but trick the webview into thinking that the user is always watching the video even if the app is in the background.

I'd appreciate this feature being added as an opt in toggle. Thanks.

**Comment by golgothan666** [comment]
i think the better feature would be to make YT think you watched the ADs

have it think you're watching the video at 2x while it displays the normal AD breaks, as GrayJay downloads and buffers it to your device, so you don't even notice (unless you're also watching at 2x)

which is kind of what i thought they were already doing, because that would kill the AD model pretty quick... but A.I. browsers, like Arc, with allow people to do the same thing, so Advertising shouldn't be around much longer

then we'll switch to Decentralized p2p sharing, with Blockchain micropayments, to fund content

**Comment by PhilC813** [comment]
For those who just want an alternative YouTube frontend and don't mind the ads, it makes a lot of sense. I personally don't mind ads much and I know that many creators depend on Google AdSense revenues.

**Comment by SarcevicAntonio** [comment]
I stumbled here because I thought it was confusing that the app supports sponsor block but shames you for using it while blocking YouTube ads by default without even the option to enable them.

**Comment by rabaimor** [comment]
I also would appreciate an option like this. Not because I want to watch ads, but I would enjoy faster load times if I'm already logged in with a premium account that wouldn't get ads anyway

**Comment by skinnyhastrup-design** [comment]
> I also would appreciate an option like this. Not because I want to watch ads, but I would enjoy faster load times if I'm already logged in with a premium account that wouldn't get ads anyway

Wouldn't you get faster load times when logged in on premium account anyways, because of the "playback blocked" disabled?

**Comment by The-Nickname** [comment]
Ideally wouldn't it be better (long-term) to build upon the watchtime in the creator tab.
Ideally
1. Watchtime would be counted.
2. Donation Links from channels/Polycentric-IDs would be extracted
3. The user would be notified about donating to a certain channel once a user definable threshold is reached
4. The user donates an amount(either manually or semi-automatically) to the creators they watch(according to watchtime or with 0.5x, 2x, 3x,... modifiers)

In practice this could look like that:
0. I set my wachtime-value to 1€ per 1h watched and notifications to every end of the month
1. I watch 10h of FUTO videos and 20h of Rossmann videos
2. FUTO has a donation Link in their Polycentric Profile, Rossmann has a donation link as well
3. Grayjay tells me at the end of the month "Hey you have watched a total of 30h accorss 2 channels, click here to donate to them"
4. I get a suggestion to donate 10€ to FUTO and 20€ to Rossmann. I keep the value for FUTO at 10€ and manually edit the value for Rossmann to 30€ since I want Clinton the cat to get some extra food. I hit transfer and autorize both donations.
(5. Grayjay marks those hours as "donated" or "paid")


In a perfect world, maybe one with Polycentric, other Apps, like SocialMedia or [Podcast clients](https://github.com/AntennaPod/AntennaPod/issues/4671) could locally count my watchtime/numbers as well and potentially aggregate watchtime for one creator accross multiple platforms(everything locally or E2EE of course).
I as end-user can than see those values and donate money etc as I see fit based on those numbers.

---
Just my 2ct to revolutionize the entire funding model of the internet, but IMHO (personalized and) manipulative ads are incompatible with democracy for obvious reasons.
Here are just 2 sources for examples:
https://blog.mozilla.org/en/firefox/firefox-privacy-protections/?#:~:text=this%20information%20will%20be%20used
https://en.wikipedia.org/wiki/Cambridge_Analytica

**Comment by SarcevicAntonio** [comment]
> Ideally wouldn't it be better (long-term) to build upon the watchtime in the creator tab.

while this is a cool idea, it's really a separate feature request what this thread is about.

**Comment by The-Nickname** [comment]
> > Ideally wouldn't it be better (long-term) to build upon the watchtime in the creator tab.
> 
> while this is a cool idea, it's really a separate feature request what this thread is about.

You are right, maybe I should have opened a new issue, although such a feature would make the current issue mostly obsolete, wouldn't it?

**Comment by PhilC813** [comment]
> > > Ideally wouldn't it be better (long-term) to build upon the watchtime in the creator tab.
> > 
> > 
> > while this is a cool idea, it's really a separate feature request what this thread is about.
> 
> You are right, maybe I should have opened a new issue, although such a feature would make the current issue mostly obsolete, wouldn't it?

Well no, because your idea requires the user to donate money. The whole idea of funding through ads is that it's free to the end user (minus your privacy ofc).

**Comment by The-Nickname** [comment]
> > > > Ideally wouldn't it be better (long-term) to build upon the watchtime in the creator tab.
> > > 
> > > 
> > > while this is a cool idea, it's really a separate feature request what this thread is about.
> > 
> > 
> > You are right, maybe I should have opened a new issue, although such a feature would make the current issue mostly obsolete, wouldn't it?
> 
> Well no, because your idea requires the user to donate money. The whole idea of funding through ads is that it's free to the end user (minus your privacy ofc).

I mean, yes and I get your point. But even excluding privacy etc, time and electricity isn't free(nor cheep in some countries).
I cannot decide if it should stray open or not, those where just my 2ct ;)
